arwen/elrondapi/bigFloatOps.go

```diff
@@ -0,0 +1,893 @@
+		return
+	}
+	if oneIsInfinity(op) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	managedType.ConsumeGasForBigFloatCopy(dest, op)
+	dest.Sqrt(op)
+}
+
+//export v1_4_bigFloatLog2
+func v1_4_bigFloatLog2(context unsafe.Pointer, opHandle int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatLog2
+	metering.UseGas(gasToUse)
+
+	op, err := managedType.GetBigFloat(opHandle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return -1
+	}
+	if op.Sign() < 0 {
+		_ = arwen.WithFault(arwen.ErrBadLowerBounds, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return -1
+	}
+	if oneIsInfinity(op) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return -1
+	}
+	bigIntOp := new(big.Int)
+	op.Int(bigIntOp)
+	managedType.ConsumeGasForBigFloatCopy(op)
+	managedType.ConsumeGasForBigIntCopy(bigIntOp)
+	return int32(bigIntOp.BitLen() - 1)
+}
+
+//export v1_4_bigFloatPow
+func v1_4_bigFloatPow(context unsafe.Pointer, destinationHandle, op1Handle, op2Handle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigIntAPICost.BigIntPow
+	metering.UseGas(gasToUse)
+
+	dest := managedType.GetBigFloatOrCreate(destinationHandle)
+	op1, op2, err := managedType.GetTwoBigFloat(op1Handle, op2Handle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return
+	}
+	if oneIsInfinity(op1, op2) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	managedType.ConsumeGasForBigFloatCopy(op1, op2, dest)
+
+	op1BigInt := big.NewInt(0)
+	op1.Int(op1BigInt)
+	op2BigInt := big.NewInt(0)
+	op1.Int(op2BigInt)
+	if op1BigInt.Sign() > 0 {
+		op1BigInt.Add(op1BigInt, big.NewInt(1))
+	}
+	if op2BigInt.Sign() > 0 {
+		op2BigInt.Add(op2BigInt, big.NewInt(1))
+	}
+
+	//this calculates the length of the result in bytes
+	lengthOfResult := big.NewInt(0).Div(big.NewInt(0).Mul(op2BigInt, big.NewInt(int64(op1BigInt.BitLen()))), big.NewInt(8))
+	managedType.ConsumeGasForThisBigIntNumberOfBytes(lengthOfResult)
+
+	dest.Set(pow(op1, op2))
+}
+
+func pow(base *big.Float, exp *big.Float) *big.Float {
+	result := new(big.Float).Copy(base)
+	counter := big.NewFloat(1)
+	basePrecision := base.Prec()
+	expPrecision := exp.Prec()
+	if basePrecision > expPrecision {
+		counter.SetPrec(basePrecision)
+		result.SetPrec(basePrecision)
+	} else {
+		counter.SetPrec(expPrecision)
+		result.SetPrec(expPrecision)
+	}
+	for counter.Cmp(exp) < 0 {
```

Comment:
This isn't how real number exponentiation works. This is float ^ int.
Rename it to powInt or something and make the second argument int32.
For real number exponentiation you would need some Exp function and combine that with Log2. No need to implement now.